File: CountingCards/bank.cpp
//
//  bank.cpp
//  CountingCards
//
//  Created by Gabe Scott on 7/25/22.
//

//#include "bank.hpp"
                                            
class Bank
{
private:
    unsigned int _balance;
    Bank();     // private default constructor - can't be called
public:
    Bank(int startBalance): _balance(startBalance) { }
    int getBalance() {return _balance; }
    void setBalance(int value) {_balance = value; }
    void addFunds( int value ) {_balance += value; }
    void removeFunds(int value) {(_balance-value<=0) ? _balance=0 : _balance-=value; }
    void payBlackjack(int bet) { _balance += (bet * 3 / 2); }
    void payWin(int bet) { _balance += bet; }
};


                                             

File: scripts/bank.js
//
//  bank.js
//  CountingCards
//
//  Copyright (c) 2023 Gabe Scott
//

class Bank {
    constructor(startBalance) {
        if(typeof(startBalance) != "number") {
            throw new Error('expected type number, got ' + typeof startBalance);
        }
        this.balance = startBalance;
        this.addNextHand = 0;
    }

    getBalance() {
        return this.balance;
    }

    addFunds(value) {
        this.balance += value;
    }

    removeFunds(value) {
        this.balance -= value;
    }

    payBlackjack(bet) {
        this.balance += (bet * 3 / 2); 
        return this.balance;
    }

    payWin(bet) {
        this.balance += bet;
    }

    getAddNextHand() {
        return this.addNextHand;
    }

    setAddNextHand(amount) {
        this.addNextHand = amount;
    }
}